ReSwift/CoreTypes/MainStore.swift

```diff
@@ -91,28 +88,23 @@ public class MainStore<State: StateType>: Store {
     }
 
     public func dispatch(action: Action, callback: DispatchCallback?) -> Any {
-        let returnValue = self.dispatchFunction(action)
-        callback?(self.state)
+        let returnValue = dispatchFunction(action)
+        callback?(state)
 
         return returnValue
     }
 
     public func dispatch(actionCreatorProvider: ActionCreator, callback: DispatchCallback?) -> Any {
-        let action = actionCreatorProvider(state: self.state, store: self)
-        if let action = action {
-            dispatch(action, callback: callback)
-        }
+        let action = actionCreatorProvider(state: state, store: self)
+        _ = action.map { dispatch($0, callback: callback) }
 
```

Comment:
Here I strongly prefer the optional chaining version over `map`. It is easier to read and using `map` for side-effects (like here) should generally be avoided.